### PR TITLE
7

### PR DESCRIPTION
**Pull request by SamenaBasith**

### CODE REVIEW

File: src/api.js [hunk 1]
```diff
@@ -23,4 +23,16 @@ export const getSingleArticle = (article_id) => {
     .then((res) => {
       return res.data.comments;
     });
-  };
\ No newline at end of file
+  };
+
+
+  export const patchArticleById = (article_id, increment) => {
+    const patchedBody = {
```
Comment (lewispricey) [inline src/api.js:30]: Really like that you've seperated the body out, keeps the code super clean!

File: src/components/ArticleVotes.jsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { useState } from "react"
+import { patchArticleById } from '../api'
+
+const ArticleVotes = ({article_id, votes}) => {
+
+    const [votesChange, setVotes]= useState(0)
+
+    function incVote(increment) {
```
Comment (lewispricey) [inline src/components/ArticleVotes.jsx:9]: Great re-use of the same function to increase and reduce votes

File: src/components/ArticleVotes.jsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        setVotes((currVote) => 
+        currVote + increment);
+        patchArticleById(article_id, increment)
+        .catch(() => {
```
Comment (lewispricey) [inline src/components/ArticleVotes.jsx]: It would be good to let the user know with some sort of error message if the vote fails

**Review by lewispricey — CHANGES_REQUESTED** [review]
Really great work on this Samena!

I'd recommend adding an error message to show the user if the like fails but overall really solid code :)

Please raise another PR once you've added in the error message.

**Comment by Dutton1990** [comment]
Yeah this looks great Samena nice job! 

If you like, you could implement only being able to vote once, but it's upto you and you don't need to send the PR again for us to check either way 😄 

You can merge this!

**Pull request merged**